### ISSUE TITLE
[Bug]: $not:null not working in filters (especially for objectbrick fields)

### ISSUE THREAD

**Issue by mkwlf** [pimcore/data-hub#483]
### Expected behavior

In the demo querying for a objectbrick field which is null works fine:
```graphql 
query GetCarsCylindersAreNull {
  getCarListing(filter : "{\"cylinders\":null}"){
    edges {
      node {
        name
        cylinders
      }
    }
  }
}
```
returns:
```json
{
  "data": {
    "getCarListing": {
      "edges": [
        {
          "node": {
            "name": "Käfer",
            "cylinders": null
          }
        },
       ... etc ...
      ]
    }
  }
}
```

When querying for the inverse the query results in an error

```graphql
# inverse query
query GetCarsCylindersNotNull {
  getCarListing(filter: "{\"cylinders\": {\"$not\":null}}") {
    edges {
      node {
        name
        cylinders
      }
    }
  }
}
```

### Actual behavior

Query
```graphql
# inverse query
query GetCarsCylindersNotNull {
  getCarListing(filter: "{\"cylinders\": {\"$not\":null}}") {
    edges {
      node {
        name
        cylinders
      }
    }
  }
}
```

returns 

```json
{
  "errors": [
    {
      "message": "An exception occurred while executing 'SELECT object_localized_CAR_en.o_id as o_id, object_localized_CAR_en.o_type as o_type FROM object_localized_CAR_en LEFT JOIN object_brick_query_Engine_CAR `Engine` ON 1\nAND `Engine`.o_id = `object_localized_CAR_en`.o_id WHERE ( (\n            (\n                SELECT `read` from `plugin_datahub_workspaces_object`\n                WHERE `plugin_datahub_workspaces_object`.configuration = 'products'\n                AND LOCATE(CONCAT(`object_localized_CAR_en`.o_path,`object_localized_CAR_en`.o_key),`plugin_datahub_workspaces_object`.cpath)=1\n                ORDER BY LENGTH(`plugin_datahub_workspaces_object`.cpath) DESC\n                LIMIT 1\n            )=1\n            OR\n            (\n                SELECT `read` from `plugin_datahub_workspaces_object`\n                WHERE `plugin_datahub_workspaces_object`.configuration = 'products'\n                AND LOCATE(`plugin_datahub_workspaces_object`.cpath,CONCAT(`object_localized_CAR_en`.o_path,`object_localized_CAR_en`.o_key))=1\n                ORDER BY LENGTH(`plugin_datahub_workspaces_object`.cpath) DESC\n                LIMIT 1\n            )=1\n        ) AND  (( NOT `object_localized_CAR_en`.`cylinders` IS NULL) )  AND  object_localized_CAR_en.o_type IN ('object','folder','variant')) AND object_localized_CAR_en.o_published = 1':\n\nSQLSTATE[42S22]: Column not found: 1054 Unknown column 'object_localized_CAR_en.cylinders' in 'where clause'"
    }
  ]
}
```
for better readability:

```sql
An exception occurred while executing 'SELECT object_localized_CAR_en.o_id as o_id, object_localized_CAR_en.o_type as o_type FROM object_localized_CAR_en LEFT JOIN object_brick_query_Engine_CAR `Engine` ON 1
AND `Engine`.o_id = `object_localized_CAR_en`.o_id WHERE ( (
            (
                SELECT `read` from `plugin_datahub_workspaces_object`
                WHERE `plugin_datahub_workspaces_object`.configuration = 'products'
                AND LOCATE(CONCAT(`object_localized_CAR_en`.o_path,`object_localized_CAR_en`.o_key),`plugin_datahub_workspaces_object`.cpath)=1
                ORDER BY LENGTH(`plugin_datahub_workspaces_object`.cpath) DESC
                LIMIT 1
            )=1
            OR
            (
                SELECT `read` from `plugin_datahub_workspaces_object`
                WHERE `plugin_datahub_workspaces_object`.configuration = 'products'
                AND LOCATE(`plugin_datahub_workspaces_object`.cpath,CONCAT(`object_localized_CAR_en`.o_path,`object_localized_CAR_en`.o_key))=1
                ORDER BY LENGTH(`plugin_datahub_workspaces_object`.cpath) DESC
                LIMIT 1
            )=1
        ) AND  (( NOT `object_localized_CAR_en`.`cylinders` IS NULL) )  AND  object_localized_CAR_en.o_type IN ('object','folder','variant')) AND object_localized_CAR_en.o_published = 1':

SQLSTATE[42S22]: Column not found: 1054 Unknown column 'object_localized_CAR_en.cylinders' in 'where clause'
```

### Steps to reproduce

Open the demo and try to run this query:
```graphql
query GetCarsCylindersNotNull {
  getCarListing(filter: "{\"cylinders\": {\"$not\":null}}") {
    edges {
      node {
        name
        cylinders
      }
    }
  }
}

**Comment by florian25686** [comment]
Can confirm this behavior with the demo project

**Comment by florian25686** [comment]
@fashxp @mcop1 FYI

**Comment by dvesh3** [comment]
Fixed by #490
